x-pack/plugins/stack_connectors/server/connector_types/crowdstrike/crowdstrike.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import type { AxiosError } from 'axios';
+import { SubActionRequestParams } from '@kbn/actions-plugin/server/sub_action_framework/types';
+import type {
+  CrowdstrikeConfig,
+  CrowdstrikeSecrets,
+  CrowdstrikeGetAgentsResponse,
+  CrowdstrikeGetAgentsParams,
+  CrowdstrikeBaseApiResponse,
+  CrowdstrikeHostActionsParams,
+  CrowdstrikeGetTokenResponse,
+} from '../../../common/crowdstrike/types';
+import {
+  CrowdstrikeGetAgentsResponseSchema,
+  CrowdstrikeHostActionsParamsSchema,
+  CrowdstrikeGetAgentsParamsSchema,
+  CrowdstrikeGetTokenResponseSchema,
+  CrowdstrikeHostActionsResponseSchema,
+} from '../../../common/crowdstrike/schema';
+import { SUB_ACTION } from '../../../common/crowdstrike/constants';
+
+export const API_PATH = 'https://api.crowdstrike.com';
+
+const paramsSerializer = (params: Record<string, string>) => {
+  return Object.entries(params)
+    .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
+    .join('&');
+};
+
+export class CrowdstrikeConnector extends SubActionConnector<
+  CrowdstrikeConfig,
+  CrowdstrikeSecrets
+> {
+  private static token: string | null;
+  private static tokenExpiryTimeout: NodeJS.Timeout;
```

Comment:
Suggestion: add some jsdocs here on these two properties explaining why we are doing this?

Reply:
Forget to add some docs?